config/routes.rb

```diff
@@ -1,9 +1,10 @@
 Rails.application.routes.draw do
   resources :vacancies
+  resources :main
```

Comment:
You are defining main as a resource while associated controller doesn't have any related actions. Why are you doing this?